### PR TITLE
rose bush: cycles add new->old/old->new button

### LINKED ISSUE #1610: rose bush: cycles list add radio button old ->new / new->old
Sort by descending already, change to be able to sort ascending depending on the radio button selected.

### PR DESCRIPTION
**Pull request by kaday**
Closes #1610

### CODE REVIEW

**Comment by kaday** [comment]
@arjclark please review

File: lib/python/rose/bush.py [hunk 1]
```diff
@@ -99,6 +101,10 @@ def cycles(self, user, suite, page=1, per_page=None, form=None):
             "host": self.host_name,
             "user": user,
             "suite": suite,
+            "cycles": cycles,
+            "is_option_on": is_option_on,
+            "tasks": tasks,
```
Comment (matthewrmshin) [inline lib/python/rose/bush.py]: I can't see how `is_option_on` and `tasks` are used.

File: lib/html/rose-bush/cycles.html [hunk 2]
```diff
@@ -106,6 +137,9 @@ <h3 id="suite-info-label">Suite Info</h3>
 <input type="hidden" name="user" value="{{user}}" />
 <input type="hidden" name="suite" value="{{suite}}" />
 <input type="hidden" name="per_page" value="{{per_page}}" />
+{% if order -%}
+<input type="hidden" name="order" value="{{order}}" />
+{% endif -%}
```
Comment (matthewrmshin) [inline lib/html/rose-bush/cycles.html]: This shouldn't be necessary?

File: lib/python/rose/suite_engine_procs/cylc.py [hunk 3]
```diff
@@ -497,9 +498,19 @@ def get_suite_cycles_summary(self, user_name, suite_name, limit, offset):
...
+            except:
+                stmt += ") GROUP BY cycle ORDER BY cast(cycle as number) DESC"
         else:
-            stmt += ") GROUP BY cycle ORDER BY cycle DESC"
+            try: 
+                (self.ORDERS.get(order)).rsplit(' ')[-1]
+                stmt += ") GROUP BY cycle ORDER BY cycle " + ( 
+                         self.ORDERS.get(order)).rsplit(' ')[-1]
+            except:
+                stmt += ") GROUP BY cycle ORDER BY cycle DESC"
```
Comment (matthewrmshin) [inline lib/python/rose/suite_engine_procs/cylc.py]: I cannot see how this works. I suspect you want the choices to be:

```
time DESC, cycle DESC
time ASC, cycle ASC
cycle DESC
cycle ASC
```

?

**Comment by matthewrmshin** [comment]
I thought I would take a look at this while @arjclark is on leave.

**Comment by arjclark** [comment]
@kaday - will await addressing Matt's feedback.

File: lib/python/rose/bush.py (hunk 1 above)
Comment (kaday) [inline lib/python/rose/bush.py]: Agree and removed, a different rose issue was previously causing these to be required and Ben fixed that yesterday so it now works without them.

File: lib/html/rose-bush/cycles.html (hunk 2 above)
Comment (kaday) [inline lib/html/rose-bush/cycles.html]: Its not, removed.

**Comment by kaday** [comment]
@arjclark - I have addressed all of Matt's feedback. If you are happy and before you merge please can I rebase it?

File: lib/html/rose-bush/index.html [hunk 4]
```diff
@@ -28,15 +28,15 @@
 </form>
 </div>
 <div class="span6">
-<form action="{{script}}/jobs" class="form-inline">
+<form action="{{script}}/jobs" class="form-inline"/>
+<form action="{{script}}/cycles" class="form-inline"/>
```
Comment (matthewrmshin) [inline lib/html/rose-bush/index.html]: This change appears to to introduce an embedded form that does not have a close tag. The original code allows the form action to change using the `formaction` attribute in the `Cycles List` button.

File: lib/python/rose/suite_engine_procs/cylc.py [hunk 5]
```diff
@@ -47,6 +47,11 @@ class CylcProcessor(SuiteEngineProcessor):
 
     """Logic specific to the Cylc suite engine."""
 
+    CYCLE_ORDERS = {
+        "time_desc":
+        " DESC",
+        "time_asc":
+        " ASC"}
```
Comment (matthewrmshin) [inline lib/python/rose/suite_engine_procs/cylc.py]: All these should fit in a single line?

**Comment by arjclark** [comment]
@matthewrmshin - at this point, given your level of passing by review and feeding back I'm assigning this to you as reviewer.

**Comment by kaday** [comment]
@matthewrmshin done let me know if you are happy with it and will rebase before you merge it. Thanks.

File: lib/html/rose-bush/index.html [hunk 6]
```diff
@@ -36,8 +36,7 @@
 <input id="suite" name="suite" type="text" class="input-small"
 placeholder="Suite ID" required="1" />
 <input type="submit" class="btn btn-primary" value="Cycles List"
-formaction="{{script}}/cycles" />
-<input type="submit" class="btn btn-primary" value="Jobs List" />
+-formaction="{{script}}/cycles" />
```
Comment (matthewrmshin) [inline lib/html/rose-bush/index.html]: And now you have deleted the "Jobs List" button.

**Comment by kaday** [comment]
Oh yes, sorted. @matthewrmshin please review

**Comment by kaday** [comment]
@matthewrmshin  button moved and you seem happier

**Pull request merged**